src/Expander/Monad.hs

```diff
@@ -184,7 +187,7 @@ data EValue
   = EPrimMacro (Ty -> SplitCorePtr -> Syntax -> Expand ()) -- ^ For special forms
   | EPrimTypeMacro (SplitTypePtr -> Syntax -> Expand ()) -- ^ For type-level special forms
   | EPrimModuleMacro (Syntax -> Expand ())
-  | EPrimDeclMacro (Scope -> DeclPtr -> DeclValidityPtr -> Syntax -> Expand ())
+  | EPrimDeclMacro (DeclTreePtr -> ScopeSet -> DeclValidityPtr -> Syntax -> Expand ())
```

Comment:
Now takes a `ScopeSet` instead of a `Scope`.

Reply:
I think using ScopeSets here was a very good decision, and it seems to be making the whole thing work out nicely.